source/ring_buffer.c

```diff
@@ -70,26 +83,34 @@ int aws_ring_buffer_acquire(struct aws_ring_buffer *ring_buf, size_t requested_s
         size_t space = tail_cpy - head_cpy - 1;
 
         if (space >= requested_size) {
-            aws_atomic_store_ptr(&ring_buf->head, head_cpy + requested_size);
+            AWS_ATOMIC_STORE_PTR(head_cpy + requested_size, ring_buf, &ring_buf->head);
```

Comment:
As I noted in the comment above: the diff would be less confusing if you kept the parameters to the macro in the same order as the parameters to the original `aws_atomic_store_ptr` function.